### PR TITLE
CMR-9207: Updating key extraction and labeling for download assets

### PR DESCRIPTION
**Pull request by dpesall**
This change will adopt the current labeling strategy for download assets, by extracting the key name from the second-to-last index of the href url when split on periods.

Current functionality:
```"assets": {
      "download_0": {
            "href": "s3://lp-prod-protected/HLSL30.020/HLS.L30.T59WPT.2013101T001445.v2.0/HLS.L30.T59WPT.2013101T001445.v2.0.B09.tif",
            "title": "Direct Download [1]",
            "description": "This link provides direct download access via S3 to the granule",
            "roles": [
                  "data"
            ]
      },
      "download_1": {
            "href": "s3://lp-prod-protected/HLSL30.020/HLS.L30.T59WPT.2013101T001445.v2.0/HLS.L30.T59WPT.2013101T001445.v2.0.Fmask.tif",
            "title": "Direct Download [1]",
            "description": "This link provides direct download access via S3 to the granule",
            "roles": [
                  "data"
            ]
      },      
}
```

New functionality:
```"assets": {
      "B09": {
            "href": "s3://lp-prod-protected/HLSL30.020/HLS.L30.T59WPT.2013101T001445.v2.0/HLS.L30.T59WPT.2013101T001445.v2.0.B09.tif",
            "title": "Direct Download [1]",
            "description": "This link provides direct download access via S3 to the granule",
            "roles": [
                  "data"
            ]
      },
      "Fmask": {
            "href": "s3://lp-prod-protected/HLSL30.020/HLS.L30.T59WPT.2013101T001445.v2.0/HLS.L30.T59WPT.2013101T001445.v2.0.Fmask.tif",
            "title": "Direct Download [1]",
            "description": "This link provides direct download access via S3 to the granule",
            "roles": [
                  "data"
            ]
      },           
}
```

Here is what it will look like if the download asset has a bad URL:

```
asset_key: {
    description: "Example of bad url data",
    href: "ftp://e4ftl015/ExampleBadUrl",
    title: "Direct Download [1]",
    roles: ["data"],
},
```

### CODE REVIEW

File: src/domains/stac.ts [hunk 1]
```diff
@@ -42,10 +42,10 @@ const downloadAssets = (concept: Collection | Granule) => {
   return (concept.relatedUrls ?? [])
     .filter((relatedUrl) => relatedUrl["type"] === RelatedUrlType.GET_DATA)
     .reduce((downloadAssets, relatedUrl, idx, available) => {
-      const [entry, display] = available.length > 1 ? [`_${idx}`, ` [${idx}]`] : ["", ""];
-
+      const display = available.length > 1 ? ` [${idx}]` : "";
```
Comment (eudoroolivares2016) [inline src/domains/stac.ts:45]: ultra nitpick but

```suggestion
      const display = available.length > 1 ? `[${idx}]` : "";
```

Comment (dpesall) [reply src/domains/stac.ts:45]: Actually this space was needed to add a space between `Direct Download` and `[1]` in `Direct Download [1]`. It'll look a bit squished.

Comment (eudoroolivares2016) [reply src/domains/stac.ts:45]: Okay its fine that way. If its for that then I would think you could just add it to `title: `Direct Download ${display}`,`?

File: src/utils/index.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ export const wrapErrorHandler = (fn: (rq: Request, rs: Response) => Promise<void
  */
 export const extractAssetMapKey = (relatedUrl: string) => {
   const urlArray = relatedUrl.split(".");
-  return urlArray[urlArray.length - 2] ? urlArray[urlArray.length - 2] : "key";
+  return urlArray[urlArray.length - 2] ? urlArray[urlArray.length - 2] : "asset_key";
```
Comment (eudoroolivares2016) [inline src/utils/index.ts:52]: nitpick but, if we are not initializing values for these sort of ternaries in the code I'm fine
```suggestion
  const assetKey = urlArray[urlArray.length - 2];
  return assetKey ? assetKey : "asset_key";
```

**Review by eudoroolivares2016 — APPROVED** [review]

**Review by daniel-zamora — APPROVED** [review]

**Pull request merged**